src/App.vue

```diff
@@ -610,23 +613,54 @@ export default {
       }
     },
 
-    /** Verifies and stores a draft applications data. */
+    /** Verifies and stores a draft application's data. */
     storeDraftApp (application: any): void {
       const filing = application?.filing
       const filingName = filing.header?.name as FilingTypes
+
       if (!filing || !filing.header || !filingName) {
         throw new Error(`Invalid ${filingName} filing`)
       }
 
-      if (![FilingTypes.INCORPORATION_APPLICATION, FilingTypes.REGISTRATION].includes(filingName)) {
-        throw new Error(`Invalid ${filingName} filing - filing name`)
-      }
-
```

Comment:
I moved this in to a switch-case statement below, where I also set the entity status. This is because we previously had an if-else (2 possible draft apps) but now we have 3.